megamek/src/megamek/client/ui/swing/CustomMechDialog.java

```diff
@@ -275,8 +285,9 @@ public CustomMechDialog(ClientGUI clientgui, Client client, List<Entity> entitie
         boolean eligibleForOffBoard = true;
         
         for (Entity e : entities) {
-            isAero &= e instanceof Aero;
+            isAero &= e instanceof Aero && !(e instanceof SmallCraft || e instanceof Jumpship);
             isMech &= e instanceof Mech;
+            isShip &= e instanceof SmallCraft || e instanceof Jumpship;
```

Comment:
Our coding convention is to put parenthesis around each value, so this should be `(e instanceof SmallCraft) || (e instanceof Jumpship)`

Reply:
Fixed.